src/Neo.VM/ReferenceCounter.cs

```diff
@@ -115,12 +115,12 @@ internal int CheckZeroReferred()
                             tracked_items.Remove(item);
                             if (item is CompoundType compound)
                             {
-                                references_count -= compound.SubItemsCount;
                                 foreach (StackItem subitem in compound.SubItems)
                                 {
                                     if (component.Contains(subitem)) continue;
                                     if (!NeedTrack(subitem)) continue;
                                     subitem.ObjectReferences!.Remove(compound);
+                                    references_count--;
```

Comment:
I'd expect references removal to be symmetric wrt references addition. And currently it's asymmetric because there's `if (!NeedTrack(subitem)) continue;` a few lines above; and during references addition we always increment refcounter irrespective of `NeedTrack`:
https://github.com/neo-project/neo/blob/b1d27f0189861167b8005a7e40e6d8500fb48cc4/src/Neo.VM/ReferenceCounter.cs#L49-L50

Reply:
Yes, maybe we need to move the line before that if

Reply:
The thing that I'm worried about is that we should find the reason why symmetric approach doesn't work. Right now I don't completely understand it.